src/uci.lua

```diff
@@ -175,7 +196,16 @@ function set_value(module)
 	end
 
 	local res
-	if not sec then
+	if nil ~= p["match"] then
+		local message = {
+			config=conf,
+			values=p["values"],
+			match=p["match"]
+		}
+		-- TODO: how can we properly report whether a changed happend in this case?
+		-- Get the entire config before and after to compare?
```

Comment:
UCI has the option to query changes (globally, or for specific sections) and exposes it via:
```
ubus call uci changes "{'config': 'network'}"
```
(you can test this via the ubus binary on your openwrt box to get a feeling of the return values)
I think this is the most promising way here, to do a diff of the changes before and after the set.
Naturally, it has to be called before the commit operation, though as we usually call `docommit` just before module_exit.
Cons: We have to write a differ for lua-tables?